src/App.jsx

```diff
@@ -1,24 +1,61 @@
-import {useEffect, useState} from 'react'
+import {useEffect, useRef, useState} from 'react'
 import { SignClient } from '@walletconnect/sign-client';
 import { Web3Modal } from '@web3modal/standalone';
 import './App.css'
 
 
 // Your dapp's Project ID from https://cloud.walletconnect.com/
-const WC_PROJECT_ID = '9e1152b9dc0318eea105dc31238fbc00';
+const WC_PROJECT_ID = 'cc3b2ecffec2ba45dfa80295a62b3f5a';
 const CHAIN_ID = 'finschia-2';
+
 // https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md
 const CAIP_BLOCKCHAIN_ID = `cosmos:${CHAIN_ID}`
-const signClient = await SignClient.init({
+
+const web3modal = new Web3Modal({
+    walletConnectVersion: 2,
     projectId: WC_PROJECT_ID,
-    metadata: {
-        name: "WC2 dApp",
-        description: "WalletConnect v2 Dapp Example for DOSI Vault Extension",
-        url: "https://dapp.example.com",
-        icons: ["https://i.pinimg.com/600x315/93/3e/14/933e14abb0241584fd6d5a31bea1ce7b.jpg"],
-    },
+    standaloneChains: [CAIP_BLOCKCHAIN_ID]
 });
-const connection = await signClient.connect({});
+
+function initSignClient() {
+    const [uri, setUri] = useState('')
```

Comment:
@tvd12 any reason to change `wcUri` to `uri`?

Reply:
sorry, it's just I like `uri` than `wcUri` with no reason :D, do you want me to change from `uri` to `wcUri`?

Reply:
ok. let's keep it as `uri` here.
but if someone changes it, we need to find out 'why' so please keep the changes minimal :)